### ISSUE TITLE
DirectoryIterator should have a sorted filenames when shuffle=False

### ISSUE THREAD

**Issue by rdelassus** [fchollet/keras#6418]
I am trying to perform images segmentation.
In one directorty I have a list of files :

- Image1.tif
- Image2.tif
- Image3.tif

And in an other directory the masks :

- Mask1.tif
- Mask2.tif
- Mask3.tif

If I say to flow_from_directory with shuffle=False or Shuffle=True but seeds are the same, I expect the files to be yielded in the same order. They are not.

A simple self.filenames.sort() juste before shuffling should solve the problem.

**Comment by JakeBecker** [comment]
The issue seems to be that `os.walk` yields filenames in arbitrary order, as mentioned in the docs for `os.listdir`: https://docs.python.org/3/library/os.html#os.listdir

On some systems like Mac OSX, the filenames are given in lexicographic order, but on Ubuntu with Ext4 filesystem, they seem to be in some other arbitrary order.

Fixing this is important for the "Example of transforming images and masks together" in the docs: https://keras.io/preprocessing/image/
